workshops/memory game/README.md

```diff
@@ -0,0 +1,555 @@
+
+It looks like this with all the cards turned down (i.e., placeholder images).
+
+![Temporary output](https://cloud-2vppbtbp1.vercel.app/0jsop.png)
+
+## Flip Card
+
+When a card is clicked, let's flip the card. 
+
+Create two empty variable arrays `cardsClicked` and `cardsClickedId`. Also create a variable `cardsMatched` array to push the matched cards
+
+```js
+var cardsClicked = []
+var cardsClickedId = []
+var cardsMatched = []
+```
+
+Create a function `flipCard()`. Then inside this function, create a variable `cardId`, which is the `data-id` attribute of the clicked card, using `getAttribute()`. 
+
+```js
+function flipCard() {
+  var cardId = this.getAttribute('data-id')
+  // code goes here
+}
+```
+
+Now add name of this card into `cardsClicked` array based on `cardId` using `push()` method. Also push the id of this card (i.e., `cardId`) into `cardsClickedId` array. 
+
+```js
+cardsClicked.push(cardArray[cardId].name)
+cardsClickedId.push(cardId)
+```
+
+Then add the front side of the card, image in `cardArray` corrresponding to `cardId`, using `setAttribute`.
+
+```js
+this.setAttribute('src', cardArray[cardId].img)
+```
+
+After selecting two cards, we have to check for match. For that, if two cards are clicked i.e., the length of `cardsClicked` array is equal to `2`, invoke `checkForMatch()` inside `setTimeout()` so that everything doesn't happen too fast.
+
+```js
+if (cardsClicked.length ===2) {
+  setTimeout(checkForMatch, 500)
+}
+```
+    
+**Explaination :** Here, 
+- [getAttribute()](https://developer.mozilla.org/en-US/docs/Web/API/Element/getAttribute) returns the value of a specified attribute on the element.
+- [push()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/push) method adds one or more elements to the end of an array and returns the new length of the array.
+- [setTimeout()](https://developer.mozilla.org/en-US/docs/Web/API/WindowOrWorkerGlobalScope/setTimeout) sets a timer which executes a function or specified piece of code once the timer expires.
+
+Final `flipCard` function will be :
+
+```js
+function flipCard() {
+  var cardId = this.getAttribute('data-id')
+  cardsClicked.push(cardArray[cardId].name)
+  cardsClickedId.push(cardId)
+  this.setAttribute('src', cardArray[cardId].img)
+  if (cardsClicked.length ===2) {
+    setTimeout(checkForMatch, 500)
+  }
+}
+```
+
+<details>
+  <summary> Here's what the code looks like so far: </summary>
+  
+```javascript
+document.addEventListener('DOMContentLoaded', () => {
+  const cardArray = [....]// the cardArray we created before
+  
+  const board = document.querySelector('.board')
+  const result = document.querySelector('#score')
+  const placeholder = "https://cloud-5ystxzer7.vercel.app/7placeholder.png"
+  const blank = "https://cloud-5ystxzer7.vercel.app/6blank.png"
+  var cardsClicked = []
+  var cardsClickedId = []
+  var cardsMatched = []
+
+  //creating game board
+    function createBoard() {
```

Comment:
shouldn't this be at the same indentation as the above comment? i think this function is indented too much by 2 spaces